src/registries/npm-registry.js

```diff
@@ -271,6 +249,7 @@ export default class NpmRegistry extends Registry {
       if (registry) {
         return String(registry);
       }
+      return '';
```

Comment:
This is to make sure that if URL is with a protocol, e.g. `https://github.com`, we don't "assume" this is `registry.npmjs.org`. This change is making `getRegistry` stricter. Since this method is used only for authentication and scope resolving purposes, we basically make sure to be strict and only return `registry.npmjs.org` if request is really to that registry.